tests/deferred_yield_higher/Makefile

```diff
@@ -0,0 +1,5 @@
+include ../Makefile.tests_common
+
+FEATURES_BLACKLIST += arch_avr8 arch_msp430 arch_riscv arch_mips32r2 arch_esp
```

Comment:
This does not seem like a good thing, this hides that the test does not pass for this archs. If the test just exposes the current state, just let the test fail.

Reply:
Then add a feature? It would also allow code that expects this behaviour to use it or guard against it better.

Reply:
I'd rather not go that route. This behavior (thread_yield_higher() defers yield when called with interrupts disabled) makes implementing some core logic simpler in some cases. Having some platforms support this and others not doesn't make sense. Unless all do, we should treat this as not supported.

Reply:
OK